default/AI/AIstate.py

```diff
@@ -662,6 +661,18 @@ def rate_fleet(self, fleet_id, enemy_stats=None, ship_info=None):
                 shields = ship.currentMeterValue(fo.meterType.shield)
                 stats['structure'] = structure
                 stats['shields'] = shields
+                stats['attack'] = 0
+                stats['attacks'] = {}
+                design = ship.design
+                if not design:
+                    print "ERROR: Ship has no valid design :)"
+                    print "If this happened in a moderated game: Screw those. The AI simply does not work with them."
+                    continue
+                for partname in design.parts:
+                    if partname and fo.getPartType(partname).partClass == fo.shipPartClass.shortRange:
+                        damage = ship.currentPartMeterValue(fo.meterType.capacity, partname)
+                        stats['attack'] += damage
+                        stats['attacks'][damage] = stats['attacks'].get(damage, 0) + 1
```

Comment:
Do we need lines 664, 665 if they are already defined? No need yo define it twice.